JASP-Engine/JASP/R/ttestindependentsamples.R

```diff
@@ -15,978 +15,589 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-TTestIndependentSamples <- function(dataset = NULL, options, perform = "run",
-																		callback = function(...) 0, ...) {
-
-	state <- .retrieveState()
-
-	## call the common initialization function
-	init <- .initializeTTest(dataset, options, perform, type = "independent-samples")
-
-	results <- init[["results"]]
-	dataset <- init[["dataset"]]
-
-	if (length(options$variables) != 0 && options$groupingVariable != '') {
-		errors <- .hasErrors(dataset, perform, type = 'factorLevels',
-												 factorLevels.target = options$groupingVariable, factorLevels.amount = '!= 2',
-												 exitAnalysisIfErrors = TRUE)
-	}
-
-	## call the specific independent T-Test functions
-	results[["ttest"]] <- .ttestIndependentSamplesTTest(dataset, options, perform)
-	descriptivesTable <- .ttestIndependentSamplesDescriptives(dataset, options, perform)
-	levene <- .ttestIndependentSamplesInequalityOfVariances(dataset, options, perform)
-	shapiroWilk <- .ttestIndependentSamplesNormalityTest(dataset, options, perform)
-	results[["assumptionChecks"]] <- list(shapiroWilk = shapiroWilk, levene = levene, title = "Assumption Checks")
-
-
-	keep <- NULL
-	## if the user wants descriptive plots, s/he shall get them!
-	if (options$descriptivesPlots && length(options$variables) > 0) {
-
-		plotTitle <- ifelse(length(options$variables) > 1, "Descriptives Plots", "Descriptives Plot")
-		descriptivesPlots <- .independentSamplesTTestDescriptivesPlot(dataset, options, perform)
-		if (!is.null(descriptivesPlots[[1]][["obj"]])){
-			keep <- unlist(lapply(descriptivesPlots, function(x) x[["data"]]),NULL)
-		}
-		results[["descriptives"]] <- list(descriptivesTable = descriptivesTable,
-																			title = "Descriptives",
-																			descriptivesPlots = list(collection = descriptivesPlots,
-																															 title = plotTitle))
-
-	} else {
-
-		results[["descriptives"]] <- list(descriptivesTable = descriptivesTable, title = "Descriptives")
-
-	}
-
-	## return the results object
-	if (perform == "init") {
-		return(list(results=results, status="inited"))
-	} else {
-		return(list(results=results, status="complete",
-								state = list(options = options, results = results),
-								keep = keep))
-	}
+TTestIndependentSamples <- function(jaspResults, dataset = NULL, options, ...) {
+  #at least one variable and one grouping variable
+  ready <- length(options$variables) > 0 && options$groupingVariable != ""
+  type  <- "independent"
+  if(ready) {
+    dataset <- .ttestReadData(dataset, options, type)
+    .ttestCheckErrors(        dataset, options, type)
+  }
+  # Output tables (each calls its own results function)
+  .ttestIndependentMainTable(  jaspResults, dataset, options, ready, type)
+  .ttestIndependentNormalTable(jaspResults, dataset, options, ready, type)
```

Comment:
The assumption check tables should appear above the descriptives. At the moment they are above descriptives in paired samples, but below descriptives in one sample and independent samples